### ISSUE TITLE
Consider license change?

### ISSUE THREAD

**Issue by StargateMC** [PC-Logix/LanteaCraft#444]
Random thing... is it possible you'd consider a license change?
Something that allows contributions..... at least...........

Otherwise you're going to find people just ditch this mod completely if no one else can maintain it...

**Comment by mRichard629** [comment]
Why can't you just make your own fork?

**Comment by StargateMC** [comment]
I have. i just really like lochies mod, and would love to be able to contribute fixes etc to it and improvements (in exchange for the rights to run modified copies of the mod).

He's also really inactive, so it would make sense for this to happen.

(it is up to him, so no one else needs to comment on this... He can close it without comment too lol)

**Comment by StargateMC** [comment]
Stargatemc.com, by the way @mrichard629

**Comment by StargateMC** [comment]
Seeing as you edited your post Richards - the license PC logix use prohibits contributions by the community - have a read.
It also prohibits "distribution of modified copies of Lanteacraft" which basically means I can't put it on a public server unless it's unmodified.

**Comment by Cronosus** [comment]
this mod really need someone to work on them and/or fix necessary things (OC, vortex, shutingdown, dialing...)
Lochie not have time for it :(

**Comment by StargateMC** [comment]
@Cronosus The license prohibits people helping.

**Comment by StargateMC** [comment]
I'm going to close this issue. Lochie would have seen it and I don't want it to become a comment thread :)

**Comment by AfterLifeLochie** [comment]
@DraksterAU, I totally get where you're coming from. To be honest, my interest in modding Minecraft using Forge has waned pretty severely for a variety of different reasons. 

I know full well that if I do decide to pick up this project again, the first task will be diving into 1.8+ and porting a lot of the actual engine which sits between the LC code and Forge. There's a lot of careful work involved in this, since most of the LC objects (blocks, tiles) are not native Forge objects and are instead abstracted through wrappers to hide weird behaviours or to make creating of Forge objects faster.

For example, there is a lot of hard-coded rendering "pipeline" stuff already to make working on 1.7.x easier, but a lot of this isn't applicable to 1.8 (block rendering, etc). It could be ported to 1.8 in a variety of ways -- you could rewrite the block-render handling, or a smarter solution would to be convert the block-rendering pipeline's handler to generate models instead and inject them into the runtime. This requires some ASM-level injection, but is easily possible in 1.8+ environments.

I would like to just make LanteaCraft a DWTFYW/OSI license, but we have some other restrictions and licensing issues which prevent us from doing so at the moment. I will have to think of a way we can work around this, while still keeping LanteaCraft up to date. This might involve us allowing pull requests and automating the builds to avoid breaching other license conditions -- I don't know yet.

TL:DR; I don't want to see the mod die, but we have licensing restrictions currently preventing us from letting anyone do whatever they want. I'm going to look into ways which we can continue the mod in an open way, but while respecting other's licenses.

**Comment by StargateMC** [comment]
Hi @AfterLifeLochie ,

Im curious, Copyright only protects you from unauthorised distribution of code, regardless of license correct?
This should mean I can modify gate behaviours etc server-side as long as the client pack connecting stores one of the versions of LanteaCraft that is 'official'?

The reason I ask is that my server (stargatemc.com) is currently undergoing a refresh of sorts, which is an opportunity for me to potentially replace SGCraft with something more .... feature rich.

Sorry for the random post, but would there be any problem (legal or otherwise) for me to run a custom build of LC on a publically available server?

This is all pending gates being 100% stable etc of course :) 

Regards,
Bysokar

**Comment by AfterLifeLochie** [comment]
Yes, this is allowed -- not for the reason you think. Our copyright applies to code and binaries; please read it carefully, but we permit you to mess with the source (build it yourself for SSP, etc) so long as you don't share it.    (EDIT: to clarify, "it" includes code and binaries, in the event this wasn't clear)

(Our licence doesn't restrict a lot of technical loopholes which you could quite easily use. Though you could use these (and I'd probably never know), it's not exactly in the spirit of things.)

Eventually, you will run into issues. If we roll out certain features which raise some security concerns, you'll have to use signed copies of the binaries to utilize them -- this isn't intended to be DRM, rather, it's a measure to make sure that LC can't be manipulated into doing nasty things. You can certainly patch around these measures, but you'll find there's some architectural designs which you can't work around. Put short -- you can compile it yourself, but don't expect support (since whatever changes you make could cause anything) and don't expect it to work with every single feature we roll out.

I definitely haven't forgotten about this issue though; I'm still considering ways we can fix this. 👍

**Comment by lieven121** [comment]
i have done quit some work on the 1.7.10 version of lanteacraft on my local machine but i cant pull request to the 1.7.10 tread
